ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/router_bgp.py

```diff
@@ -170,6 +172,41 @@ def router_bgp(self) -> dict | None:
             if neighbors:
                 router_bgp["neighbors"] = neighbors
 
+        for interface in self.shared_utils.wan_bgp_interfaces:
+            neighbor = {
+                "ip_address": get(interface, "peer_ip"),
+                "remote_as": get(interface, "bgp.peer_as"),
+                "peer_group": self.shared_utils.bgp_peer_groups["wan_underlay_peers"]["name"],
+            }
+
+            description = self.shared_utils.interface_descriptions.underlay_ethernet_interface(
+                InterfaceDescriptionData(
+                    shared_utils=self.shared_utils,
+                    peer=interface.get("peer"),
+                    peer_interface=interface.get("peer_interface"),
+                )
+            )
```

Comment:
I don't think we have peer and peer_interface available here. We need to figure out a different description style (so we cannot reuse underlay_ethernet_interface). We already have a description for the interface itself, so you can reuse that logic.

Reply:
What do you mean ?

Reply:
This is what we use for interface description

interface_description = l3_interface.get("description")
        if not interface_description:
            interface_description = self.shared_utils.interface_descriptions.underlay_ethernet_interface(
                InterfaceDescriptionData(
                    shared_utils=self.shared_utils,
                    interface=interface_name,
                    peer=l3_interface.get("peer"),
                    peer_interface=l3_interface.get("peer_interface"),
                    wan_carrier=l3_interface.get("wan_carrier"),
                    wan_circuit_id=l3_interface.get("wan_circuit_id"),
                )
            )
            
    
   Custom user description didn't make sense to me, so I removed that, and kept peer and peer_interface. I also removed the wan_carrier/circuit_id, but can add it back if u think .